fastestimator/backend/resize3d.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2022 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import List, TypeVar
+
+import tensorflow as tf
+import torch
+import torchvision
+
+Tensor = TypeVar('Tensor', tf.Tensor, torch.Tensor)
+
+
+def resize_3d(tensor: Tensor, output_shape: List[int]) -> Tensor:
+    """Reshape a `tensor` to conform to a given shape.
+
+    This method can be used with TensorFlow tensors:
+    ```python
+    t = tf.constant([[[[[0.], [1.]], [[2.], [3.]]], [[[4.], [5.]], [[6.], [7.]]]]])
+    b = fe.backend.resize_3d(t, output_shape=[3, 3, 3])  # [[[[[0.], [0.5], [1.]], [[1.], [1.5], [2.]], [[2.], [2.5], [3.]]],
+                                                      [[[2.], [2.5], [3.]], [[3.], [3.5], [4.]], [[4.], [4.5], [5.]]], [6.]], [[6.], [6.5], [7.]]]
+    ```
+
+    This method can be used with PyTorch tensors:
+    ```python
+    p = torch.tensor([[[[[0., 1.], [2., 3.]], [[4., 5.], [6., 7.]]]]])
+    b = fe.backend.resize_3d(p, output_shape=[3, 3, 3])  # [[[[[0., 0.5, 1.], [1., 1.5, 2.], [2., 2.5, 3.]],
+                                                       [[2., 2.5, 3.], [3., 3.5, 4.], [4., 4.5, 5.]],
+                                                       [[4., 4.5, 5.], [5., 5.5, 6.], [6., 6.499999, 7.]]]]]
+    ```
+
+    Args:
+        tensor: The input value.
+        output_shape: The new size of the tensor.
+
+    Returns:
+        The resized `tensor`.
+
+    Raises:
+        ValueError: If `tensor` is an unacceptable data type.
+    """
+    if tf.is_tensor(tensor):
+        return resize_tensorflow_tensor(tensor, output_shape)
+    elif isinstance(tensor, torch.Tensor):
+        return resize_pytorch_tensor(tensor, output_shape)
```

Comment:
Using interpolation in pytorch, duplicates or pads with constant values rather than actual resize. So, following same approach for both tensorflow and python.